### PR TITLE
add submodule

### PR DESCRIPTION
**Pull request by alberhander**
- Adding new submodule: https://github.com/CartoDB/cartodb-private into `private/` dir.
- Adding new Cloud Build workflow files.
- Replacing deploy key used to clone private repos.

### CODE REVIEW

File: script/ci/cloudbuild-new-build-master.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      pip3 install gitpython
+      python3 trigger_build.py
+
+# Build image from stage 1
+- name: gcr.io/cloud-builders/docker
+  entrypoint: /bin/bash
+  args:
+    - -c
+    - |
+      if [ -f /workspace/build_trigger_1 ]; then
```
Comment (ilbambino) [inline script/ci/cloudbuild-new-build-master.yaml:101]: why do we have that `if` I don't see the `build_trigger_1` being created anywhere

**Review by ilbambino — COMMENTED** [review]
why do we build the docker images for all postgres versions? If they are all triggered by a PR, shouldn't it be built only once?

File: script/ci/cloudbuild-new-build-master.yaml (hunk 1 above)
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml:86]: the file will be created https://github.com/CartoDB/cartodb-private/blob/ci_tests_passing/script/ci/trigger_build.py#L43-L46 only when the `Dockerfile.stage1` is modified.

File: script/ci/cloudbuild-new-build-master.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        docker pull gcr.io/cartodb-on-gcp-main-artifacts/builder-stage1:latest
```
[the remark this answers is not in the document]
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml:87]: yes, doing `pull` + `build --cache-from` will take advantage of cache layers

File: script/ci/cloudbuild-new-build-master.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    mv script/ci/known_hosts /root/.ssh/known_hosts
+  volumes:
+  - name: 'ssh'
+    path: /root/.ssh
+
+# Clean working directory
+- name: alpine
+  args: ['sh', '-c', 'rm -rf .* | true && rm -rf *']
+
+# Checkout repo with submodules (temp: to commit c6ce872fb8facbf046bd7bb33138194e071d61ec)
```
[the remark this answers is not in the document]
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml]: Removed!

File: script/ci/cloudbuild-new-build-master.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+# Checkout repo with submodules
+- name: 'gcr.io/cloud-builders/git'
+  entrypoint: 'sh'
+  args: ['-c', 'git clone git@github.com:CartoDB/cartodb-private.git -b "${BRANCH_NAME}" --depth 2 --recursive']
+  volumes:
+  - name: 'ssh'
+    path: /root/.ssh
+
+# Notify build start via Slack
+- name: ubuntu
```
[the remark this answers is not in the document]
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml]: Removed, we are using a different approach for that.

File: script/ci/cloudbuild-new-build-master.yaml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    - -c
+    - |
+
+# Some of this won't be necessary once everything is in the same repo
+- name: ubuntu
+  entrypoint: /bin/bash
+  args:
+    - -c
+    - |
+      cp cartodb-private/script/ci/* .
```
[the remark this answers is not in the document]
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml]: Yes, updated already.

File: script/ci/cloudbuild-new-build-master.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alberhander) [reply script/ci/cloudbuild-new-build-master.yaml:87]: yes

**Review by alberhander — COMMENTED** [review]
Added some comments

**Pull request merged**